pkg/apis/velero/v1/snapshot.go

```diff
@@ -31,36 +25,45 @@ type SnapshotSpec struct {
 
 	// BackupUID is the UID of the Velero backup this snapshot
 	// is associated with.
+	// +optional
 	BackupUID string `json:"backupUID"`
 
 	// Location is the name of the VolumeSnapshotLocation where this snapshot is stored.
+	// +optional
 	Location string `json:"location"`
 
 	// PersistentVolumeName is the Kubernetes name for the volume.
-	PersistentVolumeName string `json:persistentVolumeName`
+	// +optional
+	PersistentVolumeName string `json:"persistentVolumeName"`
```

Comment:
Would at least one of `PersistentVolumeName` or `ProviderVolumeID` have to be populated?